portfolio/src/main/webapp/jquery_functions.js

```diff
@@ -0,0 +1,31 @@
+/*
+* Cycles between  different background images, triggered by mouse click
+*/
+
+let img_arr = new Array(5);
+img_arr = ["../images/mountbierstadt.jpg", 
+           "../images/mountyale.jpg",
+           "../images/gardenofgods.jpg",
+           "../images/lakedillon.jpg",
+           "../images/mountelbert.jpg"
+           ];
+labels = ["the summit of Mount Bierdstadt.", 
+          "the summit of Mount Yale.", 
+          "the Garden of the Gods.", 
+          "Lake Dillon, seen from the vantage point of driving along Dillon Dam road.",
+          "the summit of Mount Elbert, the tallest point in Colorado."];
+
+
+$(document).ready(function(){
+    // if change background button is clicked, background image for the body is changed. 
+    // Appropriate image label is displayed.
+  let current_image_index = 0;
+  $("#change-background").click(function(){
+      current_image_index++;
```

Comment:
I was trying to figure out why this was confusing to me. I think it's because when we enter the loop with index=0, we actually show image 1. And we enter the loop with index=img_arr.length, we actually show image 0. So the order of the images in the list doesn't match the index.

What you're really trying to do is treat the array as circular. Why not use index = (index + 1) mod img_arr.len? Assuming we don't click the button integer.MAX times, it should work and be more clear what's happening.

Reply:
That does sound clearer, done.

Reply:
This would even work on integer.MAX times if the modulo is being used. I think the main issue is that the `current_image_index++;` is at the top of the method, but it can just be moved to the bottom :)

Reply:
@DirtGrubDylan I moved the increment to the bottom of the method, hopefully everything is good now!

Reply:
Great!